src/components/UNSTABLE_Pagination/Pagination.js

```diff
@@ -0,0 +1,247 @@
+          tooltipPosition="top"
+          iconDescription={backwardText}
+        />
+        <Button
+          className={classnames(
+            `${namespace}__button`,
+            `${namespace}__button--forward`,
+            {
+              [`${namespace}__button--no-index`]: forwardButtonDisabled,
+            }
+          )}
+          onClick={() => setCurrentPage(currentPage + 1)}
+          disabled={forwardButtonDisabled}
+          hasIconOnly
+          renderIcon={CaretRight24}
+          tooltipAlignment="center"
+          tooltipPosition="top"
+          iconDescription={forwardText}
+        />
+      </div>
+    </section>
+  );
+}
+
+UNSTABLE_Pagination.defaultProps = {
+  backwardText: 'Previous page',
+  className: null,
+  children: undefined,
+  disabled: false,
+  forwardText: 'Next page',
+  id: 1,
+  itemsPerPageText: 'Items per page:',
+  itemRangeText: (min, max, total) => `${min}–${max} of ${total} items`,
+  itemText: (min, max) => `${min}–${max} items`,
+  page: 1,
+  pageRangeText: (current, total) => `${current} of ${total} pages`,
+  pageSize: 10,
+  pageSizes: undefined,
+  pageText: page => `page ${page}`,
+  pagesUnknown: false,
+};
+
+UNSTABLE_Pagination.propTypes = {
+  /**
+   * The description for the backward icon.
+   */
+  backwardText: PropTypes.string,
+
+  /**
+   * The children of the pagination component.
+   */
+  children: PropTypes.node,
+
+  /**
+   * Extra classes to add.
+   */
+  className: PropTypes.string,
+
+  /**
+   * `true` if the backward/forward buttons should be disabled.
+   */
+  disabled: PropTypes.bool,
+
+  /**
+   * The description for the forward icon.
+   */
+  forwardText: PropTypes.string,
+
+  /** The unique ID of this component instance. */
+  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+
+  /**
+   * The translatable text indicating the number of items per page.
+   */
+  itemsPerPageText: PropTypes.string,
+
+  /**
+   * The function returning a translatable text showing where the current page is,
+   * in a manner of the range of items.
+   */
+  itemRangeText: PropTypes.func,
+
+  /**
+   * A variant of `itemRangeText`, used if the total number of items is unknown.
+   */
+  itemText: PropTypes.func,
+
+  /**
+   * The current page.
+   */
+  page: PropTypes.number,
+
+  /**
+   * The function returning a translatable text showing where the current page is,
+   * in a manner of the total number of pages.
+   */
+  pageRangeText: PropTypes.func,
+
+  /**
+   * The number dictating how many items a page contains.
+   */
+  pageSize: PropTypes.number,
+
+  /**
+   * The choices for `pageSize`.
+   */
+  pageSizes: PropTypes.arrayOf(PropTypes.number),
+
+  /**
+   * The translatable text showing the current page.
+   */
+  pageText: PropTypes.func,
+
+  /**
+   * `true` if total number of pages is unknown.
+   */
+  pagesUnknown: PropTypes.bool,
+
+  /**
+   * The total number of items.
+   */
+  totalItems: PropTypes.number.isRequired,
```

Comment:
@SimonFinney what do you think about requiring `totalItems`? Without it, there's nothing valuable to be displayed... and it would render as `NaN` if not provided. 🤔 That change would be "breaking" but I think this would be an improvement. What do you think?

Reply:
That makes sense, and I wouldn't worry too much about it being a breaking change since this is experimental / in preview mode - I haven't dug too much into the original source, but is this line related? I'm curious as to how Carbon currently handle this:

 https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/components/Pagination/Pagination.js#L299

Reply:
The `pagesUnknown` prop isn't quite related to this... if you set that to true, then it won't show the total number of pages (like if you using a backend service) but it isn't related to how total items are treated. Here are 2 examples from their current storybook:

### `totalItems={null}` and `pagesUnknown={true}`
<img width="953" alt="Screen Shot 2019-12-17 at 8 54 54 AM" src="https://user-images.githubusercontent.com/9057921/71006505-0397d500-20ab-11ea-939d-5f8aa5910590.png">


### `totalItems={undefined}` and `pagesUnknown={true}`
<img width="1106" alt="Screen Shot 2019-12-17 at 8 50 17 AM" src="https://user-images.githubusercontent.com/9057921/71006516-08f51f80-20ab-11ea-9363-b3459cc6fa42.png">

In both cases it seems like `totalItems` is implicitly required. 🤔

Reply:
Just a thought -- if we *did* keep it so that `totalItems` isn't required, what should be shown in the component if the number isn't provided? I'm guessing we would take out the `# - # of # items` line? 🤔

Reply:
🤔 I'm unsure - my gut says the logic should be handled similarly to how Carbon currently does it, right? Although, to your point maybe, this is an opportunity for improvement

Reply:
Yeah that's what my point is above... it isn't handled in Carbon at all.

Reply:
Thanks - I thought those lines in the original `Pagination` handled the logic you were referring to, but by the sounds of it they don't!